client/modules/IDE/actions/files.js

```diff
@@ -156,6 +156,24 @@ export function updateFileName(id, name) {
   };
 }
 
+export function saveFileName(id) {
+  return (dispatch, getState) => {
+    const state = getState();
+    const selectedFile = state.files.find(file => file.id === id);
```

Comment:
i think that here, you should extract only the file name, and only `PUT` that to the API, rather than doing this server-side.